src/main/java/monday/task/TaskList.java

```diff
@@ -0,0 +1,143 @@
+package monday.task;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import monday.monday.ui.Ui;
+import monday.storage.Storage;
+
+/**
+ * The TaskList class is responsible for storing and managing the tasks in the task list.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+    private Storage storage;
+
+    /**
+     * Constructs a TaskList object with the specified file path.
+     *
+     * @param filePath the file path to store the tasks
+     */
+    public TaskList(String filePath) {
+        this.storage = new Storage(filePath);
+        try {
+            this.list = storage.load();
+        } catch (IOException | ClassNotFoundException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+    }
+
+    /**
+    * Saves the list of tasks to the storage.
+    *
+    * @throws IOException if an I/O error occurs while saving the tasks
+    */
+    private void save() {
+        try {
+            storage.save(list);
+        } catch (IOException e) {
+            System.out.println("Error saving tasks: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Adds a task to the task list.
+     *
+     * @param task The task to be added.
+     * @return Message with the task added.
+     */
+    public String addToTask(Task task) {
+        list.add(task);
+        save();
+        return Ui.addTask(task, list.size());
+    }
+
+    /**
+     * Displays the list of tasks.
+     *
+     * @return Message with the list of all tasks.
+     */
+    public String displayList() {
```

Comment:
Rather than `display`, which I believe contradicts the purpose of this method (which is to convert to a String representation), I think this is a very natural functionality to override `toString` for.

Reply:
That's a good suggestion, I will work on it.